packages/components/src/snackbar/test/index.tsx

```diff
@@ -0,0 +1,263 @@
+			'components-snackbar-explicit-dismiss'
+		);
+
+		await click( snackbar );
+
+		expect( onRemove ).not.toHaveBeenCalled();
+		expect( onDismiss ).not.toHaveBeenCalled();
+	} );
+
+	it( 'should be dismissible by clicking the close button when the `explicitDismiss` prop is set to `true`', async () => {
+		const onRemove = jest.fn();
+		const onDismiss = jest.fn();
+
+		render(
+			<Snackbar
+				explicitDismiss
+				onRemove={ onRemove }
+				onDismiss={ onDismiss }
+			>
+				Message
+			</Snackbar>
+		);
+
+		const snackbar = screen.getByTestId( testId );
+		const closeButton = within( snackbar ).getByRole( 'button', {
+			name: 'Dismiss this notice',
+		} );
+
+		await click( closeButton );
+
+		expect( onRemove ).toHaveBeenCalled();
+		expect( onDismiss ).toHaveBeenCalled();
+	} );
+
+	describe( 'actions', () => {
+		it( 'should render only the first action with a warning when multiple actions are passed', () => {
+			render(
+				<Snackbar
+					actions={ [
+						{ label: 'One', url: 'https://example.com' },
+						{ label: 'Two', url: 'https://example.com' },
+						{ label: 'Three', url: 'https://example.com' },
+					] }
+				>
+					Message
+				</Snackbar>
+			);
+
+			expect( console ).toHaveWarnedWith(
+				'Snackbar can only have one action. Use Notice if your message requires many actions.'
+			);
+
+			const snackbar = screen.getByTestId( testId );
+			const actions = within( snackbar ).getAllByRole( 'link' );
+
+			expect( actions ).toHaveLength( 1 );
+			expect( actions[ 0 ] ).toHaveTextContent( 'One' );
+		} );
+
+		it( 'should be rendered as a link when the `url` prop is set', () => {
+			render(
+				<Snackbar
+					actions={ [
+						{ label: 'View post', url: 'https://example.com' },
+					] }
+				>
+					Post updated.
+				</Snackbar>
+			);
+
+			const snackbar = screen.getByTestId( testId );
+			const link = within( snackbar ).getByRole( 'link' );
+
+			expect( link ).toHaveAttribute( 'href', 'https://example.com' );
+		} );
+
+		it( 'should be rendered as a button and call `onClick` when the `onClick` prop is set', async () => {
+			const onClick = jest.fn();
+
+			render(
+				<Snackbar actions={ [ { label: 'View post', onClick } ] }>
+					Post updated.
+				</Snackbar>
+			);
+
+			const snackbar = screen.getByTestId( testId );
+			const button = within( snackbar ).getByRole( 'button', {
+				name: 'View post',
+			} );
+
+			await click( button );
+
+			expect( onClick ).toHaveBeenCalled();
+		} );
+
+		it( 'should be rendered as a link when the `url` prop and the `onClick` are set', () => {
+			render(
+				<Snackbar
+					actions={ [
+						{
+							label: 'View post',
+							url: 'https://example.com',
+							onClick: () => {},
+						},
+					] }
+				>
+					Post updated.
+				</Snackbar>
+			);
+
+			const snackbar = screen.getByTestId( testId );
+			const link = within( snackbar ).getByRole( 'link' );
+			expect( link ).toBeVisible();
+		} );
+	} );
+
+	describe( 'useSpokenMessage', () => {
```

Comment:
This is [the same as testing the Notice component](https://github.com/WordPress/gutenberg/blob/03796669a7b2449ca44cb32049ae9950d6dfffd0/packages/components/src/notice/test/index.tsx#L61).  We are testing the `useSpokenMessage` hook itself, and if it has been tested with the `Notice` component, there might be no need to test it again here.